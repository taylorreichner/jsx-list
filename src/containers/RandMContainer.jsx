import React, {Component} from 'react';
import CharacterList from '../components/characters/CharacterList';
import getCharacters from '../services/RandMApi';

export default class RandMContainer extends Component {
    state = {
        loading: true,
        characters: [],
    };

    async componentDidMount() {
        const characters = await getCharacters();
        this.setState({
            characters,
            loading: false,
        })
    }

    render() {
        const {loading, characters } = this.state;
        if (loading) return <h1>Loading...</h1>
        return <CharacterList characters={characters} />
    }
}