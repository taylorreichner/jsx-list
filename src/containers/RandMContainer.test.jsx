import React from 'react';
import { render, screen } from '@testing-library/react';
import RandMContainer from './RandMContainer';
import { MemoryRouter } from 'react-router';

describe('RickAndMortyContainer', () => {
  it('renders a list of characters to the page', async () => {
    
    
    render(<MemoryRouter> <RandMContainer /> 
    </MemoryRouter>
    );

    screen.getByText('Loading...');

    const ul = await screen.findByRole('list', { name: 'characters' });
    expect(ul).toMatchSnapshot();
  });
});
