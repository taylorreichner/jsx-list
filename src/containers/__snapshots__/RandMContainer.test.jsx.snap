// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RickAndMortyContainer renders a list of characters to the page 1`] = `
<ul
  aria-label="characters"
>
  <li>
    <a
      href="/1"
    >
       
      <img
        alt="Rick Sanchez"
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
      <p>
        Rick Sanchez
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/2"
    >
       
      <img
        alt="Morty Smith"
        src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
      />
      <p>
        Morty Smith
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/3"
    >
       
      <img
        alt="Summer Smith"
        src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
      />
      <p>
        Summer Smith
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/4"
    >
       
      <img
        alt="Beth Smith"
        src="https://rickandmortyapi.com/api/character/avatar/4.jpeg"
      />
      <p>
        Beth Smith
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5"
    >
       
      <img
        alt="Jerry Smith"
        src="https://rickandmortyapi.com/api/character/avatar/5.jpeg"
      />
      <p>
        Jerry Smith
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/6"
    >
       
      <img
        alt="Abadango Cluster Princess"
        src="https://rickandmortyapi.com/api/character/avatar/6.jpeg"
      />
      <p>
        Abadango Cluster Princess
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/7"
    >
       
      <img
        alt="Abradolf Lincler"
        src="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
      />
      <p>
        Abradolf Lincler
      </p>
      <p>
        unknown
      </p>
    </a>
  </li>
  <li>
    <a
      href="/8"
    >
       
      <img
        alt="Adjudicator Rick"
        src="https://rickandmortyapi.com/api/character/avatar/8.jpeg"
      />
      <p>
        Adjudicator Rick
      </p>
      <p>
        Dead
      </p>
    </a>
  </li>
  <li>
    <a
      href="/9"
    >
       
      <img
        alt="Agency Director"
        src="https://rickandmortyapi.com/api/character/avatar/9.jpeg"
      />
      <p>
        Agency Director
      </p>
      <p>
        Dead
      </p>
    </a>
  </li>
  <li>
    <a
      href="/10"
    >
       
      <img
        alt="Alan Rails"
        src="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
      />
      <p>
        Alan Rails
      </p>
      <p>
        Dead
      </p>
    </a>
  </li>
  <li>
    <a
      href="/11"
    >
       
      <img
        alt="Albert Einstein"
        src="https://rickandmortyapi.com/api/character/avatar/11.jpeg"
      />
      <p>
        Albert Einstein
      </p>
      <p>
        Dead
      </p>
    </a>
  </li>
  <li>
    <a
      href="/12"
    >
       
      <img
        alt="Alexander"
        src="https://rickandmortyapi.com/api/character/avatar/12.jpeg"
      />
      <p>
        Alexander
      </p>
      <p>
        Dead
      </p>
    </a>
  </li>
  <li>
    <a
      href="/13"
    >
       
      <img
        alt="Alien Googah"
        src="https://rickandmortyapi.com/api/character/avatar/13.jpeg"
      />
      <p>
        Alien Googah
      </p>
      <p>
        unknown
      </p>
    </a>
  </li>
  <li>
    <a
      href="/14"
    >
       
      <img
        alt="Alien Morty"
        src="https://rickandmortyapi.com/api/character/avatar/14.jpeg"
      />
      <p>
        Alien Morty
      </p>
      <p>
        unknown
      </p>
    </a>
  </li>
  <li>
    <a
      href="/15"
    >
       
      <img
        alt="Alien Rick"
        src="https://rickandmortyapi.com/api/character/avatar/15.jpeg"
      />
      <p>
        Alien Rick
      </p>
      <p>
        unknown
      </p>
    </a>
  </li>
  <li>
    <a
      href="/16"
    >
       
      <img
        alt="Amish Cyborg"
        src="https://rickandmortyapi.com/api/character/avatar/16.jpeg"
      />
      <p>
        Amish Cyborg
      </p>
      <p>
        Dead
      </p>
    </a>
  </li>
  <li>
    <a
      href="/17"
    >
       
      <img
        alt="Annie"
        src="https://rickandmortyapi.com/api/character/avatar/17.jpeg"
      />
      <p>
        Annie
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/18"
    >
       
      <img
        alt="Antenna Morty"
        src="https://rickandmortyapi.com/api/character/avatar/18.jpeg"
      />
      <p>
        Antenna Morty
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/19"
    >
       
      <img
        alt="Antenna Rick"
        src="https://rickandmortyapi.com/api/character/avatar/19.jpeg"
      />
      <p>
        Antenna Rick
      </p>
      <p>
        unknown
      </p>
    </a>
  </li>
  <li>
    <a
      href="/20"
    >
       
      <img
        alt="Ants in my Eyes Johnson"
        src="https://rickandmortyapi.com/api/character/avatar/20.jpeg"
      />
      <p>
        Ants in my Eyes Johnson
      </p>
      <p>
        unknown
      </p>
    </a>
  </li>
</ul>
`;
